src/ray/pubsub/publisher.h

```diff
@@ -54,6 +56,27 @@ class EntityState {
   /// Gets the current set of subscribers, keyed by subscriber IDs.
   const absl::flat_hash_map<SubscriberID, SubscriberState *> &Subscribers() const;
 
+ protected:
+  // Subscribers of this entity.
+  // The underlying SubscriberState is owned by Publisher.
+  absl::flat_hash_map<SubscriberID, SubscriberState *> subscribers_;
+};
+
+/// State for an entity does not have total size cap on published messages.
+class BasicEntityState : public EntityState {
+ public:
+  bool Publish(const rpc::PubMessage &pub_message) override;
+
+ private:
+  std::weak_ptr<rpc::PubMessage> message_;
+};
+
+/// State for an entity that streams published messages to subscribers, with total size
+/// cap.
+class StreamEntityState : public EntityState {
```

Comment:
I think "Stream" is not really the right word here since they both send streams to the subscribers. Maybe something like "CappedEntityState" vs "BufferedEntityState"?

Also, could you update the comment to explain what happens when we exceed the cap and to explain when Basic vs Streamed should be used/why we have two different kinds?

Reply:
Good point. Renamed to `CappedEntityState` and added comment on their differences.